### ISSUE TITLE
Error Code 1: Myelin (Final synchronize failed (716))

### ISSUE THREAD

**Issue by YouSenRong** [NVIDIA/TensorRT#2776]
## Description
When I  run my model on TensorRT 8.5.3 with FP16 enabled, I get the "[runner.cpp::execute::718] Error Code 1: Myelin (Final synchronize failed (716))". When I run this model on TensorRT 8.4.3 with FP16 enable, the results are all 1. Without FP16 enabled, i.e. in FP32 mode, the inference give the correct results.

![image](https://user-images.githubusercontent.com/21186538/225838366-89f76ca2-db5a-4abe-aa3e-4bb0408708d8.png)

Besides, in build time, I get the warning as follows:
![image](https://user-images.githubusercontent.com/21186538/225838519-298b8ea6-0095-4ac2-b7f1-bcc66c2cecc8.png)

## Environment

**TensorRT Version**: 8.5.3, 8.4.3
**NVIDIA GPU**: T4
**NVIDIA Driver Version**: 460.91.03
**CUDA Version**: 11.6
**CUDNN Version**: 8.4
**Operating System**: ubuntu16.04, docker container: ubuntu20.04, 
**Python Version (if applicable)**: 
**Tensorflow Version (if applicable)**: 
**PyTorch Version (if applicable)**: 
**Baremetal or Container (if so, version)**:

**Comment by zerollzeng** [comment]
Can provide a reproduce for this? Thanks!

Would be good if you can try TRT 8.6 EA first.

**Comment by YouSenRong** [comment]
Thanks for your response.
Sorry for that I can not shared the onnx file. I am trying to find out which node leads to the misaligned problem. 
(At present, I find out that may be the Unsqueeze op, i.e. Shuffle layer in FP16 mode.)
I will have a try on TRT 8.6. Thanks.

**Comment by YouSenRong** [comment]
After I set the subnormal FP16 layer to FP32 with setPrecision(), in TensorRT 8.4.3 the results are correct, however there are still misaligned error in TensorRT 8.5.3. And I guess that it is caused by the Unsqueeze op, i.e. shuffle layer (not very sure currently).
![image](https://user-images.githubusercontent.com/21186538/226182211-8b620b0d-c86c-4232-bb36-ee33abe6fc7f.png)

**Comment by QingXIA233** [comment]
@zerollzeng @YouSenRong Hello, I ran into the same problem as mentioned above. 

**Problem description:**
1. I have this onnx model, when I use `trtexec` command to export it to tensorrt engine in FP16 mode with TensorRT , everything works fine, the engine was generated without any trouble. I even checked the outputs of trt and torch models, they are consistent, only some precision loss wrt the fp16 mode.
![image](https://github.com/NVIDIA/TensorRT/assets/64100948/1772d439-373e-4d3b-a041-3250190dac81)

2. However, when I try to use this trt model to perform model inference with C++:
![90b0dfc91b637d0e4c781a6f48b8cc6](https://github.com/NVIDIA/TensorRT/assets/64100948/b4e9e78a-94fe-41cc-9c3d-c31d7cf89ebd)
![94d3967cbb99a2d1980cb6760d81271](https://github.com/NVIDIA/TensorRT/assets/64100948/70904f67-93f9-4995-a262-82570c422eff)
 it gets error, **code: 716, reason:misaligned address**:
 ![d2e5fc8cc7f0eddae5af3a0476413bc](https://github.com/NVIDIA/TensorRT/assets/64100948/84b20e2b-0a64-4059-89fc-e7568dd3b273)
 
 3. Then I used cuda-memcheck to figure out the exact error location, and I get this log. if the log file is required, I could provide:
![image](https://github.com/NVIDIA/TensorRT/assets/64100948/38eeaf8d-d4ef-4fae-a35c-d022a067ffa9)

**Environment**:
torch: 1.8.0+cu101
onnx: 1.11.0
TensorRT: 8.6.0.1 (CUDA 11.4)

@pbridger @aaronp24 @ttyio Could anyone please give me some help? If so, I would provide the onnx and trt model files, the C++ script and error log. Please! :(

**Comment by QingXIA233** [comment]
BTW, I also changed to use enqueueV3 plugin instead of enqueueV2 according to the official TensorRT8.6 documents, but it still failed.

**Comment by YouSenRong** [comment]
> BTW, I also changed to use enqueueV3 plugin instead of enqueueV2 according to the official TensorRT8.6 documents, but it still failed.

I have also encountered the "cuda misaligned" when I run one of our models based on both TRT8.6 and TRT9.1 in FP16, but well in FP32.
I have not idea why yet, and I am trying to figure out that.
Maybe you could have a try on TRT8.4, it run well for me in FP16 with correct result. (except some performance degradation😂).
@QingXIA233

**Comment by QingXIA233** [comment]
> > BTW, I also changed to use enqueueV3 plugin instead of enqueueV2 according to the official TensorRT8.6 documents, but it still failed.
> 
> I have also encountered the "cuda misaligned" when I run one of our models based on both TRT8.6 and TRT9.1 in FP16, but it works well in FP32 without any modification. I have not idea why yet, and I am trying to figure out that. Maybe you could have a try on TRT8.4, it run well for me in FP16 with correct result. (except some performance degradation😂). @QingXIA233

Yeah, me too. when I use TensorRT8.2, both fp32 and fp16 trt models work totally well. However, when I switched to TensorRT8.6, only fp32 trt model could work, fp16 trt model failed no matter what I tried. The good thing is I could provide my onnx and trt models for them to reproduce, hope this will lead us to get the solution for this kind of problem.

**Comment by QingXIA233** [comment]
> > BTW, I also changed to use enqueueV3 plugin instead of enqueueV2 according to the official TensorRT8.6 documents, but it still failed.
> 
> I have also encountered the "cuda misaligned" when I run one of our models based on both TRT8.6 and TRT9.1 in FP16, but it works well in FP32 without any modification. I have not idea why yet, and I am trying to figure out that. Maybe you could have a try on TRT8.4, it run well for me in FP16 with correct result. (except some performance degradation😂). @QingXIA233

Yeah, me too. when I use TensorRT8.2, both fp32 and fp16 trt models work totally well. However, when I switched to TensorRT8.6, only fp32 trt model could work, fp16 trt model failed no matter what I tried. The good thing is I could provide my onnx and trt models for them to reproduce, hope this will lead us to get the solution for this kind of problem.

**Comment by YouSenRong** [comment]
Maybe you could also have a try even batch-size.
I found that the even num batch-size can work in FP16, but odd num batch-size trigger "cuda misaligned".

**Comment by ttyio** [comment]
@YouSenRong  do you have any customer plugins, because cuda is async I am not sure if this is delayed from the plugin error.

Also have tried the TRT 9.1 in OSS container? If there is no customer plugin, let's check if this is fixed issue in 9.1, thanks!
https://github.com/NVIDIA/TensorRT/tree/release/9.1

**Comment by YouSenRong** [comment]
> Also have tried the TRT 9.1 in OSS container?

Yes, I had tried TRT 9.1, but the problem still exist. There is no custom plugin in our program.

**Comment by zerollzeng** [comment]
Anyone can provide a reproduce so that we can investigate?

**Comment by ttyio** [comment]
I will close this one, and reopen after we get a repro to investigate, thanks all!
